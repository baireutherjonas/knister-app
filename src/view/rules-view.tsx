function RuleRow({label, value}:{label: string, value: string}) {
    return <div className="flex flex-row gap-2">
        <div className="flex-5">{label}</div>
        <div>=</div>
        <div className="flex-1">{value}</div>
    </div>
}

export function RulesView() {

    return <div className="flex flex-col gap-4">
        <div className="flex flex-col">
            <RuleRow label="2er" value="1"/>
            <RuleRow label="3er" value="3"/>
            <RuleRow label="4er" value="6"/>
            <RuleRow label="5er" value="10"/>
        </div>

        <div className="flex flex-col">
            <RuleRow label="2er + 2er" value="3"/>
            <RuleRow label="Full House" value="8"/>
            <RuleRow label="Str. mit 7" value="8"/>
            <RuleRow label="Str. ohne 7" value="12"/>
        </div>
    </div>
}