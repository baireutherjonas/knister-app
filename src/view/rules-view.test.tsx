import { render } from "@testing-library/react"
import { describe, expect, it } from "vitest"
import { RulesView } from "./rules-view"

describe('RulesView', () => {
    it('render and match snapshot', () => {
        const result = render(<RulesView />)

        expect(result).toMatchSnapshot()
    })
})