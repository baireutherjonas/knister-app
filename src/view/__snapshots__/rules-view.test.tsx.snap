// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`RulesView > render and match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex flex-col gap-4"
      >
        <div
          class="flex flex-col"
        >
          <div
            class="flex flex-row gap-2"
          >
            <div
              class="flex-5"
            >
              2er
            </div>
            <div>
              =
            </div>
            <div
              class="flex-1"
            >
              1
            </div>
          </div>
          <div
            class="flex flex-row gap-2"
          >
            <div
              class="flex-5"
            >
              3er
            </div>
            <div>
              =
            </div>
            <div
              class="flex-1"
            >
              3
            </div>
          </div>
          <div
            class="flex flex-row gap-2"
          >
            <div
              class="flex-5"
            >
              4er
            </div>
            <div>
              =
            </div>
            <div
              class="flex-1"
            >
              6
            </div>
          </div>
          <div
            class="flex flex-row gap-2"
          >
            <div
              class="flex-5"
            >
              5er
            </div>
            <div>
              =
            </div>
            <div
              class="flex-1"
            >
              10
            </div>
          </div>
        </div>
        <div
          class="flex flex-col"
        >
          <div
            class="flex flex-row gap-2"
          >
            <div
              class="flex-5"
            >
              2er + 2er
            </div>
            <div>
              =
            </div>
            <div
              class="flex-1"
            >
              3
            </div>
          </div>
          <div
            class="flex flex-row gap-2"
          >
            <div
              class="flex-5"
            >
              Full House
            </div>
            <div>
              =
            </div>
            <div
              class="flex-1"
            >
              8
            </div>
          </div>
          <div
            class="flex flex-row gap-2"
          >
            <div
              class="flex-5"
            >
              Str. mit 7
            </div>
            <div>
              =
            </div>
            <div
              class="flex-1"
            >
              8
            </div>
          </div>
          <div
            class="flex flex-row gap-2"
          >
            <div
              class="flex-5"
            >
              Str. ohne 7
            </div>
            <div>
              =
            </div>
            <div
              class="flex-1"
            >
              12
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex flex-col gap-4"
    >
      <div
        class="flex flex-col"
      >
        <div
          class="flex flex-row gap-2"
        >
          <div
            class="flex-5"
          >
            2er
          </div>
          <div>
            =
          </div>
          <div
            class="flex-1"
          >
            1
          </div>
        </div>
        <div
          class="flex flex-row gap-2"
        >
          <div
            class="flex-5"
          >
            3er
          </div>
          <div>
            =
          </div>
          <div
            class="flex-1"
          >
            3
          </div>
        </div>
        <div
          class="flex flex-row gap-2"
        >
          <div
            class="flex-5"
          >
            4er
          </div>
          <div>
            =
          </div>
          <div
            class="flex-1"
          >
            6
          </div>
        </div>
        <div
          class="flex flex-row gap-2"
        >
          <div
            class="flex-5"
          >
            5er
          </div>
          <div>
            =
          </div>
          <div
            class="flex-1"
          >
            10
          </div>
        </div>
      </div>
      <div
        class="flex flex-col"
      >
        <div
          class="flex flex-row gap-2"
        >
          <div
            class="flex-5"
          >
            2er + 2er
          </div>
          <div>
            =
          </div>
          <div
            class="flex-1"
          >
            3
          </div>
        </div>
        <div
          class="flex flex-row gap-2"
        >
          <div
            class="flex-5"
          >
            Full House
          </div>
          <div>
            =
          </div>
          <div
            class="flex-1"
          >
            8
          </div>
        </div>
        <div
          class="flex flex-row gap-2"
        >
          <div
            class="flex-5"
          >
            Str. mit 7
          </div>
          <div>
            =
          </div>
          <div
            class="flex-1"
          >
            8
          </div>
        </div>
        <div
          class="flex flex-row gap-2"
        >
          <div
            class="flex-5"
          >
            Str. ohne 7
          </div>
          <div>
            =
          </div>
          <div
            class="flex-1"
          >
            12
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
